tron/core/actionrun.py

```diff
@@ -373,18 +383,35 @@ def _done(self, target, exit_status=0):
             return self.machine.transition(target)
 
     def retry(self):
+        """Invoked externally (via API) when action needs to be re-tried
+        manually.
+        """
         if self.retries_remaining is None or self.retries_remaining <= 0:
             self.retries_remaining = 1
 
         if self.is_done:
             return self.fail(self.exit_status)
         else:
-            log.info("Killing the current action run for a retry")
+            log.info(f"Killing action run {self.id} for a retry")
             return self.kill(final=False)
 
+    def start_after_delay(self):
+        log.info(f"Resuming action run {self.id} after backoff")
+        self.in_delay = None
+        self.start()
+
     def restart(self):
+        """Used by `fail` when action run has to be re-tried."""
         self.machine.reset()
```

Comment:
why? i think "scheduled" represents state of action in delay fairly accurately. without adding a separate state - it's probably the best we got?

Reply:
Hmm, in my opinion, "running" makes more sense. Even if the command isn't currently running, the action has started. Otherwise, you see a job/action that already have a start time, but are still in the "scheduled" state.

Reply:
i guess it's a matter of preference, i've updated the logic to keep action in running state. everything still works.